src/MySqlConnector/Core/SqlParser.cs

```diff
@@ -265,6 +270,10 @@ public void Parse(string sql)
 				states |= FinalParseStates.NeedsNewline;
 				state = beforeCommentState;
 			}
+			else if (state == State.SingleQuotedStringSingleQuote)
+			{
+				state = State.Statement;
+			}
```

Comment:
This is a general class of error I can push a fix for (before this PR is merged).

Reply:
5060d9d5c5267cb7480b299bed4bc948a3dcad03; will merge to `master` soon.